### PR TITLE
Dashboards Revamp

### PR DESCRIPTION
**Pull request by sonamgupta21**
# Description
Summarize the change.

Fixes #<issue-number> (link all the GitHub issues this addresses)

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [ ] I have self-reviewed this PR.
- [ ] I have removed all print-debugging and commented-out code that should not be merged.
- [ ] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: static/css/siglens.css [hunk 1]
```diff
@@ -3027,22 +3033,23 @@ float: right;
     width: 100%;
     height: 100%;
     position: relative;
-    padding-left: 0px;
+    background-color: #F9FAFB;
```
Comment (nkunal) [inline static/css/siglens.css:3036]: can we use names here (will make it easier when we switch themes)

File: static/css/siglens.css [hunk 2]
```diff
@@ -3055,38 +3062,44 @@ float: right;
 
 
 .panel div.panel-header .dropdown-btn{
-    background-image: url(../assets/up-arrow-btn-dark-theme.svg);
-    width: 10px;
-    height: 6px;
+    /* background-image: url(../assets/up-arrow-btn-dark-theme.svg); */
```
Comment (nkunal) [inline static/css/siglens.css:3065]: remove commented code

File: static/css/siglens.css [hunk 3]
```diff
@@ -3055,38 +3062,44 @@ float: right;
...
+    display: inline-block;
+    margin-bottom: 1px;
+    background-size: cover;
 }
 
-[data-theme='dark'] .panel div.panel-header .dropdown-btn{
+/* [data-theme='dark'] .panel div.panel-header .dropdown-btn{
     background-image: url(../assets/up-arrow-btn-dark-theme.svg);
-}
+} */
```
Comment (nkunal) [inline static/css/siglens.css:3077]: remove commented code

File: static/css/siglens.css [hunk 4]
```diff
@@ -3055,38 +3062,44 @@ float: right;
...
 
-.panel div.panel-header .dropdown-btn.active{
+/* .panel div.panel-header .dropdown-btn.active{
     transform: rotate(180deg);
-}
+} */
```
Comment (nkunal) [inline static/css/siglens.css:3081]: remove commented code

File: static/css/siglens.css [hunk 5]
```diff
@@ -3055,38 +3062,44 @@ float: right;
...
 }
 
 .panel {
-    width: 49%;
-    height: 250px;
-    background-color: var(--datatable-bg-color);
-    border-radius: 5px;
+    width: 100%;
+    height: 100%;
+    /* background-color: var(--datatable-bg-color); */
```
Comment (nkunal) [inline static/css/siglens.css:3090]: remove commented code

File: static/js/dashboard-charts.js [hunk 6]
```diff
@@ -420,9 +420,9 @@ function displayBigNumber(value, panelId, dataType, panelIndex) {
 			$(".unit").css("font-size","55px");
 		}
 	}
-	if(panelId !== -1) {
-		resizePanelFontSize(panelIndex, panelId);
-	}
+	// if(panelId !== -1) {
+	// 	resizePanelFontSize(panelIndex, panelId);
+	// }
```
Comment (nkunal) [inline static/js/dashboard-charts.js:425]: remove commented code

File: static/js/dashboard.js [hunk 7]
```diff
@@ -36,8 +36,8 @@ $(document).ready(async function () {
     $('#new-dashboard').css("transform", "translate(170px)")
     $('#new-dashboard').css("width", "calc(100% - 170px)")
 
-    panelContainer = document.getElementById('panel-container');
-    panelContainerWidthGlobal = panelContainer.offsetWidth-215;
+    // panelContainer = document.getElementById('panel-container');
+    // panelContainerWidthGlobal = panelContainer.offsetWidth-215;
```
Comment (nkunal) [inline static/js/dashboard.js:40]: remove commented code

File: static/js/dashboard.js [hunk 8]
```diff
@@ -91,14 +151,14 @@ $(document).mouseup(function (e) {
...
-    }
-});
+// window.addEventListener('resize', function (event) {
+//     if ($('.panelEditor-container').css('display') === 'none'){
+//         panelContainerWidthGlobal = panelContainer.offsetWidth-97;
+//         recalculatePanelWidths();
+//         displayPanels();
+//         resetPanelLocationsHorizontally();
+//     }
+// });
```
Comment (nkunal) [inline static/js/dashboard.js:161]: remove commented code

File: static/js/dashboard.js [hunk 9]
```diff
@@ -114,11 +174,11 @@ $(`.dbSet-textareaContainer .copy`).click(function() {
...
-function recalculatePanelWidths(){
-    localPanels.map(localPanel => {
-        localPanel.gridpos.w = localPanel.gridpos.wPercent * panelContainerWidthGlobal;
-    })
-}
+// function recalculatePanelWidths(){
+//     localPanels.map(localPanel => {
+//         localPanel.gridpos.w = localPanel.gridpos.wPercent * panelContainerWidthGlobal;
+//     })
+// }
```
Comment (nkunal) [inline static/js/dashboard.js:181]: remove commented code

File: static/js/dashboard.js [hunk 10]
```diff
@@ -194,14 +255,17 @@ function handlePanelView() {
...
-    $('#add-panel-btn').hide();
-    $('#viewPanel-container').css('display', 'flex');
-    $('#viewPanel-container').css('height', '100%');
+    // $('#app-container').show();
+    // $('.panView-goToDB').css('display', 'block');
+    // $('#viewPanel-container').show();
+    // $('#panel-container').hide();
+    $('.panelEditor-container').show();
+    $('.popupOverlay').addClass('active');
+    // $('#app-container').hide();
```
Comment (nkunal) [inline static/js/dashboard.js:264]: remove commented code

File: static/js/dashboard.js [hunk 11]
```diff
@@ -301,96 +364,98 @@ function handleDescriptionTooltip(panelId,description,searchText) {
...
+//             let yTop = localPanels[indices[j]].gridpos.y;
+
+//             if ((xTop >= xDown && xTop <= xDown + wDown) || (xTop + wTop >= xDown && xTop + wTop <= xDown + wDown) || (xTop <= xDown && xTop + wTop >= xDown + wDown)) {
+//                 ymax = Math.max(ymax, yTop + hTop);
+//             }
+//         }
+
+//         localPanels[indices[i]].gridpos.y = ymax + 10;
+//     }
+// }
```
Comment (nkunal) [inline static/js/dashboard.js:434]: remove commented code

File: static/js/dashboard.js [hunk 12]
```diff
@@ -301,96 +364,98 @@ function handleDescriptionTooltip(panelId,description,searchText) {
...
+    // let boundaryY = localPanels[duplicatedPanelIndex].gridpos.y + localPanels[duplicatedPanelIndex].gridpos.h;
+    // for (let i = 0; i < localPanels.length; i++) {
+    //     if (localPanels[i].panelIndex == localPanels.length - 1) continue; // this is the newly created duplicate panel.
+    //     if (localPanels[i].gridpos.y >= boundaryY) // if any panel starts after the ending Y-cordinate of the duplicated panel, then it should be shifted downwards
+    //         localPanels[i].gridpos.y += localPanels[duplicatedPanelIndex].gridpos.h + 20;
+    // }
+    // resetPanelLocationsVertically();
+    // resetPanelLocationsHorizontally();
+    // resetPanelContainerHeight();
+    // displayPanelsWithoutRefreshing();
```
Comment (nkunal) [inline static/js/dashboard.js:458]: remove commented code

File: static/js/dashboard.js [hunk 13]
```diff
@@ -520,43 +571,116 @@ function updateTimeRangeForPanel(panelIndex) {
...
+            y: parseInt(localPanel.gridpos.x)
+        });
+        
+        // Append panel layout to the new grid-stack-item
+        var panelDiv = $("<div>").append(panelLayout).addClass("panel temp").attr("id", `panel${idpanel}`).attr("panel-index", localPanel.panelIndex);
+        newItem.firstChild.appendChild(panelDiv[0]);
+
+        // Rest of your code for handling panel interactions
+        // handleDrag(idpanel);
+        // handleResize(idpanel);
```
Comment (nkunal) [inline static/js/dashboard.js:662]: remove commented code

File: static/js/dashboard.js [hunk 14]
```diff
@@ -625,93 +749,98 @@ function displayPanels() {
...
+    //             runPanelAggsQuery(localPanel.queryData, localPanel.panelId, localPanel.chartType, localPanel.dataType, localPanel.panelIndex, localPanel.queryRes);
+    //         else
+    //             runPanelAggsQuery(localPanel.queryData, localPanel.panelId, localPanel.chartType, localPanel.dataType, localPanel.panelIndex);
+    //     } else
+    //         allResultsDisplayed--;
+    // })
+    // if(allResultsDisplayed === 0) {
+    // }
+    // resetPanelContainerHeight();
+// }
```
Comment (nkunal) [inline static/js/dashboard.js:842]: remove commented code

File: static/js/dashboard.js [hunk 15]
```diff
@@ -827,63 +956,63 @@ function getDashboardId() {
...
+//             $(unit).css('margin-left', `8px`);
+
+//         }
+//         if (unitSize > 85)
+//             unitSize = 85;
+//         $(unit).css('font-size', `${unitSize}px`);
+//     } else {
+//         $('.big-number-display-container .big-number').css('font-size', `180px`);
+//     }
+// }
```
Comment (nkunal) [inline static/js/dashboard.js:1015]: remove commented code

File: static/js/dashboard.js [hunk 16]
```diff
@@ -1135,56 +1263,55 @@ function addPanel(chartIndex) {
...
+    // handlePanelView();
     handlePanelEdit();
     handlePanelRemove(idpanel);
     handlePanelDuplicate();
-    handleDrag(idpanel);
-    handleResize(idpanel);
-    $(`#panel${idpanel}`).get(0).scrollIntoView({ behavior: 'smooth' });
+    // handleDrag(idpanel);
+    // handleResize(idpanel);
+    // $(`#panel${idpanel}`).get(0).scrollIntoView({ behavior: 'smooth' });
```
Comment (nkunal) [inline static/js/dashboard.js:1277]: remove commented code

File: static/js/dashboard.js [hunk 17]
```diff
@@ -1196,35 +1323,44 @@ function addDuplicatePanel(panelToDuplicate) {
...
+//     let panelContainer = document.getElementById('panel-container');
+//     panelContainer.style.minHeight = panelContainerMinHeight + 50 + "px";
+// }
+
+// window.onbeforeunload = function () {
+//     if (!flagDBSaved) {
+//         return "Unsaved panel changes will be lost if you leave the page, are you sure?";
+//     }
+//     else return;
+// };
```
Comment (nkunal) [inline static/js/dashboard.js:1363]: remove commented code

File: static/js/edit-panel-screen.js [hunk 18]
```diff
@@ -270,9 +270,38 @@ $(document).ready(function () {
...
+        $('#viewPanel-container').show();
+		$('#edit-button').removeClass('active');
+		$('#overview-button').addClass('active');
+		displayPanelView(panelIndex)
+
+    });
 })
 
 function editPanelInit(redirectedFromViewScreen) {
+	console.log("redirectedFromViewScreen",redirectedFromViewScreen);
```
Comment (nkunal) [inline static/js/edit-panel-screen.js:293]: remove log

File: static/js/edit-panel-screen.js [hunk 19]
```diff
@@ -491,7 +520,11 @@ $('.panEdit-save').on("click",async function(redirectedFromViewScreen){
     }
 	localPanels[panelIndex] = JSON.parse(JSON.stringify(currentPanel));
 	await updateDashboard();
-	applyChangesToPanel(redirectedFromViewScreen)
+	// applyChangesToPanel(redirectedFromViewScreen)
```
Comment (nkunal) [inline static/js/edit-panel-screen.js:523]: remove commented code

File: static/js/dashboard.js [hunk 20]
```diff
@@ -24,7 +24,7 @@ let timeRange = "Last 1 Hr";
 let dbRefresh ="";
 let panelContainer;
 let panelContainerWidthGlobal;
-let curFocus;
+// let curFocus;
```
Comment (nkunal) [inline static/js/dashboard.js:27]: remove commented out code

File: static/js/dashboard.js [hunk 21]
```diff
@@ -520,43 +448,108 @@ function updateTimeRangeForPanel(panelIndex) {
...
 
+// Event listener for Gridstack resize and drag events
+grid.on('change', function(event, items) {
+    items.forEach(function(item) {
+        // Find the panel in localPanels array using its ID
+        let panelIndex = localPanels.findIndex(panel => panel.panelId === item.el.id);
+        if (panelIndex !== -1) {
+            // Update the position and size of the panel in localPanels array
+            localPanels[panelIndex].gridpos.x = item.y;
+            localPanels[panelIndex].gridpos.y = item.x;
```
Comment (nkunal) [inline static/js/dashboard.js:459]: we are swapping the x, y, is this intentional or a typo ?

File: static/js/dashboard.js [hunk 22]
```diff
@@ -1124,7 +889,7 @@ function addPanel(chartIndex) {
             "w": panelWidth,
             "x": panelLeft,
             "y": panelTop,
-            "wPercent": panelWidthPercentage,
+            // "wPercent": panelWidthPercentage,
```
Comment (nkunal) [inline static/js/dashboard.js:892]: remove commented out code

**Comment by nkunal** [comment]
do we need to keep the dashboard_original file or we can remove it ?

**Review by nkunal — APPROVED** [review]

**Comment by sonamgupta21** [comment]
> do we need to keep the dashboard_original file or we can remove it ?

Yes, we can remove it. I will do that.

**Comment by sunitakawane** [comment]
on small screen , bottom part is overlapping the buttons
Screenshot 2024-05-21 at 2 39 57 PM

File: static/js/dashboard.js (hunk 21 above)
Comment (sonamgupta21) [reply static/js/dashboard.js:446]: Yeah, that is intentional.

**Review by nkunal — APPROVED** [review]

**Pull request merged**